agent/agent.go

```diff
@@ -58,14 +58,21 @@ func HaltVM(err error) {
 
 // Initialize a new agent to facilitate communications with the host
 func NewAgent(ctx context.Context, cancelF context.CancelFunc) (*Agent, error) {
-	metadata, err := GetMachineMetadata()
+	var metadata *agentapi.MachineMetadata
+	var err error
+
+	if os.Getenv(nexEnvSandbox) == "false" {
+		metadata, err = GetMachineDataFromEnv()
```

Comment:
Calling it "data" here and "metadata" below. Should probably call this `GetMachineMetadataFromEnv()` so it matches `GetMachineMetadata()`.

Reply:
agreed. VScode's automatic "suggestion" here burned me.